### ISSUE TITLE
[ENH] consul discovery backend, add support for multiple consul nodes

### ISSUE THREAD

**Issue by david-guenault** [docker/swarm#372]
With a consul "cluster" we do not know at join time which of the consul nodes are healthy. Should it be possible to specify more than one consul ip address so when joining it take the first available and healthy node ?

**Comment by david-guenault** [comment]
Also should it be possible for the consul discovery to retrieve the consul members list so if a consul node fail the agent is able to connect to the next available node ?

**Comment by vieux** [comment]
@david-guenault you can do `consul://1.1.1.1,2.2.2.2,3.3.3.3/path`

**Comment by david-guenault** [comment]
@vieux i'm not sure about that. First because it throw me an error and then because reading the consul backend code i do not see any mention of a split addr allowing to try multiple nodes (like etcd or zookeeper does). 

Here is an exemple (i have a three node running consul cluster)

```
docker -H tcp://node1:2375 exec consul-bootstrap consul members
Node              Address          Status  Type    Build  Protocol
consul-bootstrap  10.10.0.5:8301   alive   server  0.4.1  2
consul-node2      10.10.0.22:8301  alive   server  0.4.1  2
consul-node3      10.10.0.21:8301  alive   server  0.4.1  2
```

I then launch swarm join on one node with the syntax you are giving (consul://ip1,ip2/path)

```
docker -H tcp://node1:2375 run -ti --rm --name swarm-agent-node1 swarm:latest join consul://10.10.0.5:8500,10.10.0.21:8500/swarm --addr 10.10.0.5:2375 --hb 10
FATA[0000] Put http://10.10.0.5:8500,10.10.0.21:8500/v1/kv/swarm/: dial tcp: too many colons in address 10.10.0.5:8500,10.10.0.21:8500
```

The put query is wrong **PUT http://10.10.0.5:8500,10.10.0.21:8500/v1/kv/swarm/**

When launched with only one ip it works fine by the way.

```
docker -H tcp://node1:2375 run -ti --rm --name swarm-agent-node1 swarm:latest join consul://10.10.0.5:8500/swarm --addr 10.10.0.5:2375 --hb 10
INFO[0000] Registering on the discovery service every 10 seconds...  addr=10.10.0.5:2375 discovery=consul://10.10.0.5:8500/swarm
```

And the membership is ok querying each of the consul nodes with swarm list

```
swarm list consul://node1:8500/swarm
10.10.0.22:2375
10.10.0.5:2375
127.0.0.1:2375

swarm list consul://node2:8500/swarm
10.10.0.22:2375
10.10.0.5:2375
127.0.0.1:2375

swarm list consul://node3:8500/swarm
10.10.0.22:2375
10.10.0.5:2375
127.0.0.1:2375
```

**Comment by david-guenault** [comment]
Just finished to test and started some code to add support for multi consul node support ...

**Comment by vieux** [comment]
@david-guenault thanks !

**Comment by david-guenault** [comment]
@vieux ok everything seems to work (just added a loop to iterate over declared consul nodes until finding one available, since consul api does not provide an availibility check or multinode connection support i had to make http requests directly to consul http api and so added a dependency on net/http module). This is my first try with golang so i'm not sure i'm doing it the right way. I will submit the pr tomorrow.

**Comment by aluzzardi** [comment]
Thanks @david-guenault 

This will be addressed by #583
